user_tools/pyproject.toml

```diff
@@ -84,7 +84,9 @@ dependencies = [
     # dependency of shap, python [3.9, 3.12]
     "scikit-learn==1.5.2",
     # used for retrieving available memory on the host
-    "psutil==6.1.1"
+    "psutil==6.1.1",
+    # pyspark for distributed computing
+    "pyspark==3.5.0"
```

Comment:
Is it set to `3.5.0` because we assume that the default spark-runtime is `3.5.0`? `onPrem` we use spark-3.4.2 to avoid a bug with json parsing in 3.5.0..
I wonder what is the best version to use here...
perhaps we do not need to stricly set to 3.5.0? and leave it as `pyspark>=3.4.2`; or restrict it to avoid 4.00 
 `pyspark<4.00`

Reply:
Thanks Ahmed. Modified this to use `"pyspark>=3.4.2,<4.0.0"`. Tested the latest changes in the CI job.